### PR TITLE
chore: upgrade versions and dependencies

### PR DESCRIPTION
**Pull request by j-luong**
This PR updates the minimum Jenkins version to the current LTS and updates any required dependencies. This should reduce the number of critical and high SCA vulnerabilities detected by Snyk

### New Requirements
* Plugin requires Java 11
* Plugin requires Jenkins 2.401.3+

Tested and working using a Docker Jenkins instance built on `jenkins/jenkins:lts`:
Screenshot 2023-08-22 at 12 31 43
Screenshot 2023-08-22 at 12 32 08

### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: .github/build.sh [hunk 1]
```diff
@@ -4,6 +4,7 @@ SCRIPT_RELATIVE_DIR=$(dirname "${BASH_SOURCE[0]}")
 pushd "$SCRIPT_RELATIVE_DIR"/.. || exit 1
   # use local .m2 repo to speed up build
   rsync -crlDv ~/.m2 ./.m2
-  docker build -t jenkins-snyk -f .github/Dockerfile .
+  # --platform linux/amd64 is required for M1 Macs
+  docker build --platform linux/amd64 -t jenkins-snyk -f .github/Dockerfile .
```
Comment (bastiandoetsch) [inline .github/build.sh:8]: Why would you need to build it for amd64 on an arm64? Isn't arm64 nowadays supported by openjdk & docker?

File: pom.xml [hunk 2]
```diff
@@ -83,13 +88,30 @@
       <version>5.0.11</version>
       <scope>test</scope>
     </dependency>
+    <!-- fix transitive dependency conflict -->
+    <dependency>
+      <groupId>org.jenkins-ci.main</groupId>
+      <artifactId>jenkins-test-harness</artifactId>
+      <version>2034.v41c9cb_349299</version>
+      <scope>test</scope>
+      <exclusions>
```
Comment (bastiandoetsch) [inline pom.xml:97]: Where does the included test-harness come from? And why is it a conflict?

**Review by bastiandoetsch — APPROVED** [review]

**Comment by bastiandoetsch** [comment]
Please update the changelog :)

File: pom.xml (hunk 2 above)
Comment (j-luong) [reply pom.xml]: Updating `org.jenkins-ci.plugins.plugin` to the latest version (4.71) fixed this issue.

File: .github/build.sh (hunk 1 above)
Comment (j-luong) [reply .github/build.sh:8]: Right, there's [still](https://docs.docker.com/desktop/troubleshoot/known-issues/) some issues however. In this case, the scan wouldn't run unless I specified the platform.

**Pull request merged**